### PR TITLE
feat: add new sample, clickhouse-operator

### PR DESCRIPTION
**Pull request by howieyuen**
new sample support, clickhouse-operator, including:
- base: common config
- crd: crd dependent
- prod: env config

raw yaml detail: https://raw.githubusercontent.com/Altinity/clickhouse-operator/master/deploy/operator/clickhouse-operator-install-bundle.yaml

### CODE REVIEW

File: appops/clickhouse-operator/prod/main.k [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+    # spec.template.spec.containers[1:], sidecars
+    sidecarContainers = [
+        s.Sidecar{
+            name = "metrics-exporter"
+            image = "altinity/metrics-exporter:0.19.2"
+            resource = ""
+        }
+    ]
+    schedulingStrategy.resource = Undefined
```
Comment (Peefy) [inline appops/clickhouse-operator/prod/main.k:17]: remove duplicated `schedulingStrategy.resource = Undefined`

File: appops/clickhouse-operator/prod/main.k [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+# The application configuration in stack will overwrite 
+# the configuration with the same attribute in base.
+appConfiguration: frontend.Server {
+    # spec.template.spec.containers[0], main container
+    image = "altinity/clickhouse-operator:0.19.2"
+
+    # spec.template.spec.containers[1:], sidecars
+    sidecarContainers = [
+        s.Sidecar{
```
Comment (Peefy) [inline appops/clickhouse-operator/prod/main.k:11]: use `kcl-fmt` to format the kcl code. `s.Sidecar{` -> `s.Sidecar {`

File: appops/clickhouse-operator/base/rbac.k [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        {
+            apiGroups = ["apiextensions.k8s.io"]
+            resources = ["customresourcedefinitions"]
+            verbs = ["get", "list"]
+        }
+    ]
+}
+
+clusterrolebinding: rbacv1.ClusterRoleBinding {
+    __settings__: {output_type = "STANDALONE"}
```
Comment (Peefy) [inline appops/clickhouse-operator/base/rbac.k:92]: U can move the attribute into the schema definition.

File: appops/clickhouse-operator/base/rbac.k [hunk 4]
```diff
@@ -0,0 +1,111 @@
+import base.pkg.kusion_kubernetes.api.rbac.v1 as rbacv1
+
+clusterrole: rbacv1.ClusterRole {
+    __settings__: {output_type = "STANDALONE"}
```
Comment (Peefy) [inline appops/clickhouse-operator/base/rbac.k:4]: U can move the attribute into the schema definition.

**Review by markliby — APPROVED** [review]

**Review by ldxdl — APPROVED** [review]
LGTM

**Pull request merged**